payments/stripe/stripe.go

```diff
@@ -101,7 +118,43 @@ func (s *stripePaymentProvider) charge(token string, amount uint64, currency str
 	return ch.ID, nil
 }
 
-func (s *stripePaymentProvider) NewRefunder(ctx context.Context, r *http.Request) (payments.Refunder, error) {
+func (s *stripePaymentProvider) chargePaymentIntent(paymentMethodID string, amount uint64, currency string, order *models.Order, invoiceNumber int64) (string, error) {
+	params := &stripe.PaymentIntentParams{
+		PaymentMethod: stripe.String(paymentMethodID),
+		Amount:        stripe.Int64(int64(amount)),
+		Currency:      stripe.String(currency),
+		Description:   stripe.String(fmt.Sprintf("Invoice No. %d", invoiceNumber)),
+		Shipping:      prepareShippingAddress(order.ShippingAddress),
+		Params: stripe.Params{
+			Metadata: map[string]string{
+				"order_id":       order.ID,
+				"invoice_number": fmt.Sprintf("%d", invoiceNumber),
+			},
+		},
+		ConfirmationMethod: stripe.String(string(
+			stripe.PaymentIntentConfirmationMethodManual,
+		)),
+		Confirm: stripe.Bool(true),
+	}
+	intent, err := s.client.PaymentIntents.New(params)
+	if err != nil {
+		return "", err
+	}
+
+	if intent.Status == stripe.PaymentIntentStatusRequiresAction {
+		return intent.ID, payments.NewPaymentPendingError(map[string]interface{}{
+			"payment_intent_secret": intent.ClientSecret,
```

Comment:
So I'd cut this. I think that anything `_secret` makes me nervous. Like is it going to leak to the customer? logs? is that ok? 

If you cut this then I think that you can also simplify the `PaymentPendingError` b/c this is the only place I saw the `metadata` field used.

Reply:
This is actually required behaviour for the new Stripe flow to work.
This "secret" is supposed to be passed to the client. It is not being saved to the database with the transaction because I disabled that field using `sql:"-"`

I'm using the metadata field so the payment provider impl stays generic.

See:
- https://stripe.com/docs/payments/payment-intents/web#passing-to-client
- https://stripe.com/docs/payments/payment-intents/web-manual#creating-with-manual-confirmation